src/models/StyledComponent.js

```diff
@@ -186,7 +186,7 @@ class StyledComponent extends Component<*> {
 
       if (isFunction(resolvedAttrDef)) {
         // $FlowFixMe
-        resolvedAttrDef = resolvedAttrDef(context);
+        resolvedAttrDef = resolvedAttrDef({ ...props, theme });
```

Comment:
I don't think this is. Each successive call of attrs should receive the full execution context, including any change made by a previous `attrs` invocation.

Reply:
The problem is that the context seems to be constructed from the Child to the Parent, which leads to "unexpected" results. see: https://github.com/styled-components/styled-components/issues/2246#issuecomment-443435142

Reply:
I see what you mean but with your code:
```js
const Parent = styled.div.attrs(props => ({
  color: 'second',
}))``
const Child = styled(Parent).attrs(props => ({
  color: props.color + 'ary',
}))``
```
`<Child />` `color` was actually `secondary`, but for me, it should work the other way around.
static `attrs` are overridden by a Child, I don't see why it shouldn't be the case with dynamic `attrs`.